install.sh

```diff
@@ -117,6 +117,16 @@ escape_dq() {
     printf "%s" "$@" | sed -e 's/"/\\"/g'
 }
 
+# --- format version in a way that is comparable by the shell; this assumes the current versioning scheme
+pad_version() {
+    printf "%03d%03d%03d%03d%n" $(echo "${1}-999" | tr '.[:punct:][:alpha:]' ' ') 2>/dev/null
```

Comment:
It would be better to use single quoted **'**%03d%03d%03d%03d%n**'** instead of "%03d%03d%03d%03d%n" here, since there are no variables being expanded.